src/ui/components/WaitForReduxSlice.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { ReactNode } from "react";
```

Comment:
This might be a bit more abstraction than is needed for an early bail out but curious to hear more from you about it. If we want to keep it, I'd suggest converting it to a higher-order component so the usage is a bit more straightforward.

For example:

```jsx
const ConnectedDevTools = connector(waitForSlice('messages', _DevTools));
```

Reply:
The issue I'm trying to solve here is that the setup logic for the devtools is asynchronous, so we need a bail out mechanism for the DevTools component while this setup is running. That mechanism needs to be in a separate component wrapping the DevTools component because the DevTools component itself tries to read data from the redux store that is only available after the asynchronous setup logic has finished. In the end this was the simplest solution I could think of.
I haven't worked with higher-order components yet, I'd love to try that out.

Reply:
Here's a quick branch to show what I think it could look like:

https://github.com/RecordReplay/devtools/compare/master...ryanjduffy:ryan/slice-hoc?expand=1